.github/workflows/build.yml

```diff
@@ -1,25 +1,49 @@
 name: build
 
 on:
-  # Trigger analysis when pushing in main or pull requests, and when creating
-  # a pull request. 
   push:
+    paths:
+      # specific folder locations
+      - "deeprankcore/**"
+      - "tests/**"
+      # exclude filetypes
+      - "!**.md"
+      - "!**.rst"
+      - "!**.ipynb"
+      # self
+      - ".github/workflows/build.yml"
+      # setup
+      - ".github/actions/install-python-and-package/action.yml"
+      - "setup.cfg"
     branches:
       - main
   pull_request:
-      types: [opened, synchronize, reopened]
-
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths:
+      # specific folder locations
+      - "deeprankcore/**"
+      - "tests/**"
+      # exclude filetypes
+      - "!**.md"
+      - "!**.rst"
+      - "!**.ipynb"
+      # self
+      - ".github/workflows/build.yml"
+      # setup
+      - ".github/actions/install-python-and-package/action.yml"
+      - "setup.cfg"
 
 jobs:
   build:
+    if: github.event.pull_request.draft == false
     name: Build for (${{ matrix.python-version }}, ${{ matrix.os }})
     runs-on: ${{ matrix.os }}
 
     strategy:
       fail-fast: false
       matrix:
         os: ['ubuntu-latest']
-        python-version: ['3.9', '3.10']
+        python-version: ['3.9']
```

Comment:
Newest python is actually 3.11 :). But let's stick to 3.9 and 3.10 for now, I think.
We could do 3.9 for the normal build and 3.10 for the coveralls build. There is currently 1 command present in the normal build that isn't in coveralls, I can add that one to the coveralls to basically make it an extension of normal build.

Reply:
Oh and it was released in October, so many months ago. Then in the early future we should think about adding 3.11 as well, and at that point we can evaluate how critical is to remove the 3.9 version. 

For the building, I get that coveralls does the build itself so you want to test 3.9 with the build file and 3.10 with the coveralls file, because both builds are tested this way. But this is not the cleaner way of doing it, because these yml files are thought to have well separated functions and they originate the badges that we show in the readme:
<img width="495" alt="image" src="https://github.com/DeepRank/deeprank-core/assets/55382553/ba4eea40-65eb-41e1-9a66-9017eb768804">

The build should be verified separately from coveralls, which should be run only if the build jobs go well. There are ways or creating interdependent yml files and avoiding running stuff multiple times, but I didn't want to spend much time on implementing more elegant solutions and I sticked to the basic one. If you're interested of course go for it, and we can have a chat for looking together where to find such info, but otherwise I'd prefer to use the build file to actually verify the builds that we want to guarantee (3.09 and 3.10 so far), and then use one of the two versions for running coveralls.